packages/server/src/db.ts

```diff
@@ -451,23 +450,17 @@ export function isUniqueConstraintError(error: any): boolean {
 	return false;
 }
 
-const parsePostgresVersionString = (versionString: string) => {
-	// PostgreSQL 16.1 (Debian 16.1-1.pgdg120+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
-	const matches = versionString.match('PostgreSQL (.*?) ');
-	if (!matches || matches.length !== 2) throw new Error(`Cannot parse Postgres version string: ${versionString}`);
-	return matches[1];
-};
-
 export const versionCheck = async (db: DbConnection) => {
 	if (isPostgres(db)) {
 		// We only support Postgres v12+
 		// https://github.com/laurent22/joplin/issues/9695
 		// https://www.postgresql.org/docs/current/rules-materializedviews.html
-		const minPostgresVersion = '12.0';
-		const result = await db.select(db.raw('version()')).first();
-		if (result && result.version) {
-			const version = parsePostgresVersionString(result.version);
-			if (compareVersions(version, minPostgresVersion) < 0) throw new Error(`Postgres version not supported: ${result.version}. Min required version is: ${minPostgresVersion}`);
+		const minPostgresVersion = 120000;
+		const result = await db.raw('SHOW server_version_num');
```

Comment:
I don't know, it's not such an important change so I would have kept it simple and just updated the existing logic. 

Now we need to discuss this change since you've tweaked the business logic - for example it fails if the rows is an empty array. The previous code used `.first()`, which limits the need for extra checks - any reason why that was removed? Then it doesn't fail if the version can't be retrieved, but it does if it can't be parsed - should it? And are there servers that return a slightly different version number?

If you had simply added to the existing code, we could be reasonably sure that existing servers won't be broken. But since you've changed the business logic, we need to be more careful and spend more time than I'd like to going over this.

Reply:
Yes I think that would be better. I don't see any reason to change this since there's a risk it could break existing servers. A low risk, but just not worth it.